backend/app/api/api_v1/routers/documents.py

```diff
@@ -1,17 +1,43 @@
 from datetime import datetime
 from pathlib import Path
 
-from app.core.auth import get_current_active_user
 from fastapi import APIRouter, Depends, File, HTTPException, Request, UploadFile
+from sqlalchemy.exc import IntegrityError
+
+from app.core.auth import get_current_active_user, get_current_active_superuser
+from app.db.crud.document import create_document
+from app.db.schemas.document import DocumentInDB
+from app.db.schemas.metadata import DocumentCreateWithMetadata
+from app.db.session import get_db
 from navigator.core.aws import get_s3_client
 
 documents_router = r = APIRouter()
 
-# TODO return nested documents with associations
+# TODO for get_document, return nested documents with associations
 # - only show doc IDs, association types, and a hyperlink (so the associated doc can be loaded on demand)
 # - possibly a flag so nested docs can be fully hydrated?
 
 
+@r.post("/documents", response_model=DocumentInDB)
+async def post_document(
+    request: Request,
+    document_with_metadata: DocumentCreateWithMetadata,
```

Comment:
I don't understand how I create this `document_with_metadata` body for the request without creating the document first - some of the elements of `DocumentCreateWithMetadata` look like they require the document id up front.

Reply:
Well, it's commented out now... :smiley: